code/modules/food_and_drink/cakes.dm

```diff
@@ -24,59 +24,445 @@
 /obj/item/reagent_containers/food/snacks/cake/custom
 	name = "cake"
 	desc = "a cake"
-	icon_state = "cake"
+	icon_state = "cake_batter"
 	amount = 10
 	heal_amt = 2
 	use_bite_mask = 0
-	var/icing = 0
-	var/sliced = 0
-	var/icing_color =  "#FFFFFF"
 	flags = FPRINT | TABLEPASS | NOSPLASH
 	initial_volume = 100
+	w_class = 4.0
+	var/sliced = 0
+	var/list/frostingstyles = list("classic","top swirls","bottom swirls","spirals","rose spirals")
+	var/clayer = 1
+	var/amount2 //holds the amount of slices in cake 2 (not entirely, but its used for a bit of math later)
+	var/amount3 //same for cake 3
+	var/litfam //is the cake lit (candle)
+	var/datum/light/light
+
+	//Light stuffs :D
+	New()
+		..()
+		light = new /datum/light/point
+		light.set_brightness(0.8)
+		light.set_color(0.5, 0.3, 0)
+		light.attach(src)
+
+	pickup(mob/user)
+		..()
+		light.attach(user)
+
+	dropped(mob/user)
+		..()
+		SPAWN_DBG(0)
+			if (src.loc != user)
+				light.attach(src)
+
+	proc/light(var/mob/user as mob, var/message as text)
+		if (!src)
+			return
+		if (!src.litfam)
+			src.litfam = 1
+			src.damtype = "fire"
+			src.force = 3
+			light.enable()
+			if (!(src in processing_items))
+				processing_items.Add(src)
 
-	attackby(obj/item/W as obj, mob/user as mob)
-		if (istype(W, /obj/item/axe) || istype(W, /obj/item/circular_saw) || istype(W, /obj/item/kitchen/utensil/knife) || istype(W, /obj/item/scalpel) || istype(W, /obj/item/sword) || istype(W,/obj/item/saw) || istype(W,/obj/item/knife_butcher))
+			for(var/i=1,i<=src.overlays.len,i++) //searching for an unlit candle overlay, updating it, the enabling the light
+				if(src.sliced && ("[src.overlay_refs[i]]" == "slice-candle"))
+					src.ClearSpecificOverlays("[src.overlay_refs[i]]")
+					src.UpdateOverlays(new /image('icons/obj/foodNdrink/food_dessert.dmi',"slice-candle_lit"), "slice-candle_lit")
+					break
+				if("[src.overlay_refs[i]]" == "cake[src.clayer]-candle")
+					src.ClearSpecificOverlays("[src.overlay_refs[i]]")
+					src.UpdateOverlays(new /image('icons/obj/foodNdrink/food_dessert.dmi',"cake[src.clayer]-candle_lit"), "cake[src.clayer]-candle_lit")
+					break
+		return
+
+	proc/put_out(var/mob/user as mob)
+		if (!src) return
+		if (src.litfam)
+			src.litfam = 0
+			src.damtype = "brute"
+			src.force = 0
+			light.disable()
+			if (src in processing_items)
+				processing_items.Remove(src)
+		return
+
+	attackby(obj/item/W as obj, mob/user as mob) //ok this proc is entirely a mess, but its *hopfully* better on the server than the alternatives
+		var/topping //the topping the player is adding (stored as a string reference to an icon_state)
+		var/pendinglight //a variable referenced later to check if the light source on a cake needs to be updated
+		if(istype(W,/obj/item/axe) || istype(W,/obj/item/circular_saw) || istype(W,/obj/item/kitchen/utensil/knife) || istype(W,/obj/item/scalpel) || \
```

Comment:
I believe we have an item flag along the lines of CUTTING_TOOL that should be used here.